### PR TITLE
Introduce GNSS source configuration to HardwareConfig

### PR DESCRIPTION
**Pull request by lack**
This allows user configuration of GNSS as a clock source in the v2
HardwareConfig API.  This will replace the ubxcommand section in the
previous intel plugin config.

Example -> Where the old intel plugin architecture had
```yaml
    plugins:
      e810:
        ublxCmds:
          - args: #ubxtool -P 29.20 -z CFG-HW-ANT_CFG_VOLTCTRL,1
              - "-P"
              - "29.20"
              - "-z"
              - "CFG-HW-ANT_CFG_VOLTCTRL,1"
            reportOutput: false
          - args: #ubxtool -P 29.20 -e GPS
              - "-P"
              - "29.20"
              - "-e"
              - "GPS"
            reportOutput: false
          - args: #ubxtool -P 29.20 -d Galileo
              - "-P"
              - "29.20"
              - "-d"
              - "Galileo"
            reportOutput: false
          - args: #ubxtool -P 29.20 -d GLONASS
              - "-P"
              - "29.20"
              - "-d"
              - "GLONASS"
            reportOutput: false
          - args: #ubxtool -P 29.20 -d BeiDou
              - "-P"
              - "29.20"
              - "-e"
              - "BeiDou"
            reportOutput: false
          - args: #ubxtool -P 29.20 -d SBAS
              - "-P"
              - "29.20"
              - "-d"
              - "SBAS"
            reportOutput: false
          - args: #ubxtool -P 29.20 -t -w 5 -v 1 -e SURVEYIN,600,50000
              - "-P"
              - "29.20"
              - "-t"
              - "-w"
              - "5"
              - "-v"
              - "1"
              - "-e"
              - "SURVEYIN,600,50000"
            reportOutput: true
          - args: #ubxtool -P 29.20 -p MON-HW
              - "-P"
              - "29.20"
              - "-p"
              - "MON-HW"
            reportOutput: true
          - args: #ubxtool -P 29.20 -p CFG-MSG,1,38,248
              - "-P"
              - "29.20"
              - "-p"
              - "CFG-MSG,1,38,248"
            reportOutput: true
```

The equivalent in HardwareConfig v2 would be:

```yaml
 sources:
   - name: GNSS
     sourceType: gnss
     subsystem: "{subsystem}"
     boardLabel: "{boardlabel}"
     gnssConfig:
       match:
         ethernetInterface: ens2f0
       init:
         antennaVoltage: true
         constellations:
           - GPS
           - BeiDou
         survey:
           observationTime: 600
           accuracy: 50000
         extraCommands:
           - args:
               - "-p"
               - "CFG-MSG,1,38,248"
             recordOutput: true
```

It still allows an optional 'extraCommands' section to be added to the gnssConfig.init if needed, to allow adding additional arbitrary ubxtool commands, but the most common ones that users copy&paste are all accounted for in this more friendly API.

## Summary by CodeRabbit

* **New Features**
  * Added support for GNSS-based clock chain sources in HardwareConfig, including GNSS initialization, optional U-Blox command configuration, survey settings, and device matching (TTY device or Ethernet interface).
  * Extended source type options to include `gnss`.
* **Bug Fixes**
  * Updated validation to apply source-specific requirements (PTP receiver entries and GNSS configuration only when their source type is selected).
* **Tests**
  * Added table-driven validation coverage and adjusted clock-chain edge-case fixtures for DPLL.

### CODE REVIEW

File: api/v2alpha1/hardwareconfig_types.go [hunk 1]
```diff
@@ -108,14 +121,96 @@ type SourceConfig struct {
...
+	// +kubebuilder:default=true
+	AntennaVoltage bool `json:"antennaVoltage" yaml:"antennaVeltage"`
+
+	// Constellations is the list of constellations to apply
+	// +kubebuilder:validation:UniqueItems=true
+	// +kubebuilder:default={"GPS"}
+	// +optional
+	Constellations []ConstellationID `json:"constellations" yaml:"constellations"`
+
+	// SurveyIn encodes the SURVEYIN parameters to begin the initial GNSS survey at initialization
```
Comment (vitus133) [inline api/v2alpha1/hardwareconfig_types.go:179]: It's not a strong opinion, but leaving Survey as required worries me. We'll need to translate it into a specific command, and commands change between ublox versions. When changed under our feet, it will require code change in a hurry Vs. config change ( if we make it  optional)

**Review by vitus133 — COMMENTED** [review]
It looks good! one smoll comment on the required fields

File: api/v2alpha1/hardwareconfig_types.go (hunk 1 above)
Comment (lack) [reply api/v2alpha1/hardwareconfig_types.go:179]: I suppose we could use a flag value of `observationTime: 0` to mean "Don't do it"

**Review by nocturnalastro — APPROVED** [review]
/lgtm
